Library/Homebrew/postinstall.rb

```diff
@@ -0,0 +1,30 @@
+# This script is loaded by `brew postinstall` as a separate instance.
+# Thrown exceptions are propagated back to the parent process over a pipe
+
+old_trap = trap("INT") { exit! 130 }
+
+require "global"
+require "extend/ENV"
+require "debrew"
+require "fcntl"
+require "socket"
+
+TEST_TIMEOUT_SECONDS = 5*60
+
+begin
+  error_pipe = UNIXSocket.open(ENV["HOMEBREW_ERROR_PIPE"], &:recv_io)
+  error_pipe.fcntl(Fcntl::F_SETFD, Fcntl::FD_CLOEXEC)
+
+  ENV.extend(Superenv)
+  ENV.setup_build_environment
```

Comment:
I'm not sure about this. Which `ENV` should `postinstall` run under? `Superenv` or `Stdenv`?